src/protocols/bdx/BdxTransferSession.h

```diff
@@ -0,0 +1,361 @@
+     * @param maxBlockSize    The max Block size that this object supports.
+     *
+     * @return CHIP_ERROR Result of initialization. May also indicate if the TransferSession object is unable to handle this
+     *                    request.
+     */
+    CHIP_ERROR WaitForTransfer(TransferRole role, BitFlags<uint8_t, TransferControlFlags> xferControlOpts, uint16_t maxBlockSize);
+
+    /**
+     * @brief
+     *   Indicate that all transfer parameters are acceptable and prepare a SendAccept or ReceiveAccept message (depending on role).
+     *
+     * @param acceptData Data used to populate an Accept message (some fields may differ from the original Init message)
+     *
+     * @return CHIP_ERROR Result of preparation of an Accept message. May also indicate if the TransferSession object is unable to
+     *                    handle this request.
+     */
+    CHIP_ERROR AcceptTransfer(const TransferAcceptData & acceptData);
+
+    /**
+     * @brief
+     *   Reject a TransferInit message. Use Reset() to prepare this object for another transfer.
+     *
+     * @param reason A StatusCode indicating the reason for rejecting the transfer
+     *
+     * @return CHIP_ERROR The result of the preparation of a StatusReport message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR RejectTransfer(StatusCode reason);
+
+    /**
+     * @brief
+     *   Prepare a BlockQuery message. The Block counter will be populated automatically.
+     *
+     * @return CHIP_ERROR The result of the preparation of a BlockQuery message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR PrepareBlockQuery();
+
+    /**
+     * @brief
+     *   Prepare a Block message. The Block counter will be populated automatically.
+     *
+     * @param inData Contains data for filling out the Block message
+     *
+     * @return CHIP_ERROR The result of the preparation of a Block message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR PrepareBlock(const BlockData & inData);
+
+    /**
+     * @brief
+     *   Prepare a BlockAck message. The Block counter will be populated automatically.
+     *
+     * @return CHIP_ERROR The result of the preparation of a BlockAck message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR PrepareBlockAck();
+
+    /**
+     * @brief
+     *   Prepare a BlockAckEOF message. The Block counter will be populated automatically.
+     *
+     * @return CHIP_ERROR The result of the preparation of a BlockAckEOF message. May also indicate if the TransferSession object
+     *                    is unable to handle this request.
+     */
+    CHIP_ERROR PrepareBlockAckEOF();
```

Comment:
Is there a reason the state machine cannot compute the correct EOF flag state?

Reply:
I don't remember a reason why I separated them. You're right that it can be computed.